lib/takeaway.rb

```diff
@@ -0,0 +1,53 @@
+class Takeaway
+  def initialize(menu, client)
+    @menu = menu
+    @client = client
+    @menu_array = @menu.menu
+    @current_selection = nil
+  end
+
+  attr_reader :current_selection
+
+  def list_food
+    puts "*** #{@menu.name} Menu! #{@menu.logo} ***"
+    @menu_array.each do |item|
+      puts "#{1 + @menu_array.index(item)}. #{item[:name]}. (£#{'%.2f' % item[:price]})"
+    end
+    puts "*** *** *** *** *** ***"
+  end
+
+  def order(selection)
+    raise "ERROR: It looks like something went wrong with your order request." unless nil_selection?(selection)
+
+    raise "ERROR: It looks like something went wrong with your selection." unless check_total(selection)
+
+    confirm_order(selection)
+  end
+
+  def check_total(selection)
```

Comment:
You could consider creating a separate Order class for this method and your 'confirm_order' method. This would help maintain the SRP throughout. Nice use of separation of menu, takeaway and SMS though